lib/execution_segment.go

```diff
@@ -384,13 +385,82 @@ func (ess ExecutionSegmentSequence) String() string {
 	return strings.Join(result, ",")
 }
 
+func (ess ExecutionSegmentSequence) Len() int {
+	return len(ess)
+}
+
+func (ess ExecutionSegmentSequence) Less(i, j int) bool {
+	// Yes this Less is actually More, but we want it sorted in descending order and the alternative is to reverse it after sort
+	return ess[i].length.Cmp(ess[j].length) > 0
+}
+
+func (ess ExecutionSegmentSequence) Swap(i, j int) {
+	ess[i], ess[j] = ess[j], ess[i]
+}
+
+var _ sort.Interface = ExecutionSegmentSequence{}
+
+// lowest common denominator based on https://rosettacode.org/wiki/Least_common_multiple#Go
+func (ess ExecutionSegmentSequence) lcd() int64 {
+	var m, n, z big.Int
+	z = *ess[0].length.Denom()
+	for _, seg := range ess[1:] {
+		m = z
+		n = *seg.length.Denom()
+		z.Mul(z.Div(&m, z.GCD(nil, nil, &m, &n)), &n)
+	}
+
+	return z.Int64()
+}
+
 // GetStripedOffsets returns everything that you need in order to execute only
 // the iterations that belong to the supplied segment...
 //
 // TODO: add a more detailed algorithm description
-func (ess ExecutionSegmentSequence) GetStripedOffsets(segment *ExecutionSegment) (int, []int, error) {
-	start := 0
-	offsets := []int{}
-	// TODO: basically https://docs.google.com/spreadsheets/d/1V_ivN2xuaMJIgOf1HkpOw1ex8QOhxp960itGGiRrNzo/edit
-	return start, offsets, fmt.Errorf("not implemented")
+// TODO: basically https://docs.google.com/spreadsheets/d/1V_ivN2xuaMJIgOf1HkpOw1ex8QOhxp960itGGiRrNzo/edit
+func (ess ExecutionSegmentSequence) GetStripedOffsets(segment *ExecutionSegment) (int64, []int64, error) {
+	// TODO check if ExecutionSegmentSequence actually starts from 0 and goes to 1 ?
+	var matchingSegment *ExecutionSegment
+	for _, seg := range ess {
+		if seg.Equal(segment) {
+			matchingSegment = seg
+			break
+		}
+	}
+	if matchingSegment == nil {
+		return -1, nil, fmt.Errorf("missing segment %s inside segment sequence %s", segment, ess) // TODO: make a seperate error ?
```

Comment:
This would have to be validated somewhere higher up as well, after the options consolidation. This check would just be insurance, I guess, so I think there's no need